src/fourier.jl

```diff
@@ -0,0 +1,781 @@
+    elseif power_type in ["abs", "absolute"]
+        return abs.(pds)
+    else 
+        throw(ArgumentError("Incorrect Arguments"))
+    end
+
+end
+
+function bias_term(power1::T, power2::T, power1_noise::Real, 
+                   power2_noise::Real, n_ave::Integer;
+                   intrinsic_coherence::Real=1.0) where {T<:Union{AbstractVector{<:Real},Real}}
+    
+    if n_ave > 500
+        return 0.0 * power1
+    end
+    return @. power1 * power2 - intrinsic_coherence * (power1 - power1_noise) * (power2 - power2_noise) / n_ave
+end
+
+function _raw_coherence_single(cross_power::Number, power1::Real, power2::Real, 
+                               power1_noise::Real, power2_noise::Real, 
+                               n_ave::Integer; intrinsic_coherence::Real=1.0)
+
+    bsq = bias_term(power1, power2, power1_noise, power2_noise, n_ave;
+                    intrinsic_coherence=intrinsic_coherence)
+    num = real(cross_power .* conj(cross_power)) - bsq
+    if num < 0
+        num = real(cross_power * conj(cross_power))
+    end
+    den = power1 * power2
+    return num / den
+end
+
+raw_coherence(cross_power::T1, power1::T2, power2::T2, power1_noise::Real, 
+              power2_noise::Real, n_ave::Integer;
+              intrinsic_coherence::Real=1.0) where 
+              {T1<:Union{AbstractVector{<:Number},Number},T2<:Union{AbstractVector{<:Real},Real}} = 
+                    _raw_coherence_single.(cross_power, power1, power2, 
+                                       power1_noise, power2_noise, n_ave;
+                                       intrinsic_coherence)
+
+function _estimate_intrinsic_coherence_single(cross_power::Complex, power1::Real,
+                                              power2::Real, power1_noise::Real, 
+                                              power2_noise::Real, n_ave::Integer)
+    new_coherence = 1.0
+    old_coherence = 0.0
+    count = 0
+    while (!(≈(new_coherence, old_coherence, atol=0.01)) && count< 40)
+        old_coherence = new_coherence
+        bsq = bias_term(power1, power2, power1_noise, power2_noise,
+                        n_ave, intrinsic_coherence=new_coherence)
+        den = (power1 - power1_noise) * (power2 - power2_noise)
+        num = real(cross_power * conj(cross_power)) - bsq
+        if num < 0
+            num = real(cross_power * conj(cross_power))
+        end
+        new_coherence = num / den
+        count += 1
+    end
+    return new_coherence                                       
+end
+
+estimate_intrinsic_coherence(cross_power:: AbstractVector{<:Complex}, 
+                             power1:: AbstractVector{<:Real}, power2:: AbstractVector{<:Real}, 
+                             power1_noise::Real, power2_noise::Real, n_ave::Integer)= 
+                                _estimate_intrinsic_coherence_single.(
+                                    cross_power, power1, power2, 
+                                    power1_noise, power2_noise, n_ave)
```

Comment:
Is this `estimate_intrinsic_coherence` only calling `_estimate_intrinsic_coherence_single` broadcasted?

Reply:
Yeah, I think I should define a single function only and call it broadcasted everywhere, the thing was API is meant to be used with vector inputs and not just broadcasted. We can mention in the docstrings though.